system/lib/wasmfs/syscalls.cpp

```diff
@@ -425,7 +425,7 @@ static __wasi_fd_t doOpen(char* pathname,
 
 // This function is exposed to users and allows users to create a file in a
 // specific backend. An fd to an open file is returned.
-__wasi_fd_t wasmfs_create_file(char* pathname, mode_t mode, backend_t backend) {
+int wasmfs_create_file(char* pathname, mode_t mode, backend_t backend) {
```

Comment:
isn't it better to have the more explicit type here? if it doesn't match another location we'd get a link error actually, and not a silent cast. But a static assert might be better there I guess?

Reply:
The problem is that `__wasi_fd_t` is not meant to be a user-facing type, AFAICT. We certainly don't expose it via our other system libraries, at least.

Reply:
Thinking about this some more, I'm not sure what is best. Musl syscalls don't expose `__wasi_fd_t`, which is I think why you feel it should not be exposed. But the public WasmFS API could. Just like the WASI API does, for example.

I think it would be better to do so. It is the proper type, and more specific than int, and could be changed in the future easily (unlike int, which grepping for will not be useful).

OTOH, perhaps we could define a new type `wasmfs_fd_t` and use that in the WasmFS public API, so we don't depend on WASI here.

Overall I don't feel strongly, but I do think plain `int` is the least good option. I'd be fine with a TODO to investigate this later though.

Reply:
My primary motivation is to get the API to be as similar as possible to `open`, which also returns an `int`. I agree that using `int` has downsides that a more specialized type wouldn't have, though. I'll keep `int` for now and add a TODO, as you suggest. (I think that's what you had in mind, or did you mean switching back to `__wasi_fd_t` with a TODO?)